src/components/drop/DomPositionCalculator.ts

```diff
@@ -0,0 +1,183 @@
+import {CSSProperties} from 'react';
+
+export interface IDropUIPosition {
+    position: string;
+    orientation: string;
+}
+
+export const DropPodPosition = {
+    bottom: 'bottom',
+    top: 'top',
+    left: 'left',
+    right: 'right',
+};
+
+export const OrientationByPosition: {[Position: string]: string[]} = {
+    bottom: [DropPodPosition.left, DropPodPosition.right],
+    top: [DropPodPosition.left, DropPodPosition.right],
+    left: [DropPodPosition.top, DropPodPosition.bottom],
+    right: [DropPodPosition.top, DropPodPosition.bottom],
+};

```

Comment:
By typing explicitly the const, you loose auto-completion, you could simply let typescript do the work here. Also, you could reuse the DropPodPosition to make sure there is no typo.
```ts
export const OrientationByPosition = {
    [DropPodPosition.bottom]: [DropPodPosition.left, DropPodPosition.right],
    [DropPodPosition.top]: [DropPodPosition.left, DropPodPosition.right],
    [DropPodPosition.left]: [DropPodPosition.top, DropPodPosition.bottom],
    [DropPodPosition.right]: [DropPodPosition.top, DropPodPosition.bottom],
};
```